pytorch_lightning/metrics/functional/classification.py

```diff
@@ -867,11 +868,35 @@ def auroc(
                          ' target tensor contains value different from 0 and 1.'
                          ' Multiclass is currently not supported.')
 
-    @auc_decorator(reorder=True)
-    def _auroc(pred, target, sample_weight, pos_label):
-        return roc(pred, target, sample_weight, pos_label)
-
-    return _auroc(pred=pred, target=target, sample_weight=sample_weight, pos_label=pos_label)
+    # @auc_decorator(reorder=True)
+    def _auroc(pred, target, sample_weight, pos_label, max_fpr):
+        if max_fpr is None or max_fpr == 1:
+            fpr, tpr, _ = roc(pred, target, sample_weight, pos_label)
+            return auc(fpr, tpr, reorder=True)
+        if max_fpr <= 0 or max_fpr > 1:
+            raise ValueError("Expected max_fpr in range (0, 1], got: %r" % max_fpr)
+
+        fpr, tpr, _ = roc(pred, target, sample_weight, pos_label)
+        max_fpr = torch.tensor(max_fpr, device=fpr.device)
+        # torch.bucketize is slightly more performant than torch.searchsorted
+        # for 1-D arrays
+        # Add a single point at max_fpr and interpolate its tpr value
+        stop = torch.bucketize(max_fpr, fpr, out_int32=True, right=True)
```

Comment:
This restricts the PyTorch version, right? It was introduced quite recently.

Reply:
Both `torch.bucketize` or the alternative`torch.searchsorted` were added in Pytorch 1.6. Would it be better to replace them with manually written code?

Reply:
@ddrevicky I think we can keep it as it currently is and just add a version check in the beginning

Reply:
So something like this?
```python
if LooseVersion(torch.__version__) < LooseVersion('1.6.0'):
    raise RuntimeError('max_fpr argument requires torch.bucketize which is not available below PyTorch version 1.6')
```